Batteries/Data/Vector/Lemmas.lean

```diff
@@ -45,3 +46,83 @@ protected theorem ext {a b : Vector α n} (h : (i : Nat) → (_ : i < n) → a[i
   · intro i hi _
     rw [a.size_eq] at hi
     exact h i hi
+
+@[simp] theorem getElem_mk {data : Array α} {size : data.size = n} {i : Nat} (h : i < n) :
+    (Vector.mk data size)[i] = data[i] := rfl
+
+@[simp] theorem push_mk {data : Array α} {size : data.size = n} {x : α} :
+    (Vector.mk data size).push x =
+      Vector.mk (data.push x) (by simp [size, Nat.succ_eq_add_one]) := rfl
+
+@[simp] theorem pop_mk {data : Array α} {size : data.size = n} :
+    (Vector.mk data size).pop = Vector.mk data.pop (by simp [size]) := rfl
+
+@[simp] theorem getElem_push_last {v : Vector α n} {x : α} : (v.push x)[n] = x := by
+  rcases v with ⟨data, rfl⟩
+  simp
+
+-- The `simpNF` linter incorrectly claims that this lemma can not be applied by `simp`.
```

Comment:
this is a misplaced bug report

Reply:
Sorry, what do you want done with it? There's plenty more in Mathlib. :-)

It seems a bad idea to add `@[nolint simpNF]` without explanation when we know it is wrong.

Reply:
I'm saying that this should not be a `nolint`, as it's working around a bug in the linter itself. It should be a bug report, and if you have to deal with the issue in the meantime, either use `nolints.json` or `@[nolint]` with some marking that makes it clear that it is not a genuine `@[nolint]` and should be removed later